tests/unit/src/test/scala/tests/RenameSuite.scala

```diff
@@ -0,0 +1,519 @@
+    """|/a/src/main/scala/a/Main.scala
+       |package a
+       |class User{
+       |  def <<u@@nary_!>> = false
+       |}
+       |object Main{
+       |  val user = new User()
+       |  <<!>>user
+       |}
+       |""".stripMargin
+  )
+
+  same(
+    "java-classes",
+    """|/a/src/main/scala/a/Main.scala
+       |package a
+       |class MyException extends Exce@@ption
+       |class NewException extends RuntimeException
+       |class NewException2 extends RuntimeException
+       |""".stripMargin
+  )
+
+  renamed(
+    "inheritance",
+    """|/a/src/main/scala/a/Main.scala
+       |package a
+       |abstract class <<An@@imal>>
+       |class Dog extends <<Animal>>
+       |class Cat extends <<Animal>>
+       |""".stripMargin,
+    newName = "Tree"
+  )
+
+  renamed(
+    "companion",
+    """|/a/src/main/scala/a/Main.scala
+       |package a
+       |class <<Main>>{}
+       |object <<M@@ain>>
+       |""".stripMargin,
+    newName = "Tree",
+    fileRenames =
+      Map("a/src/main/scala/a/Main.scala" -> "a/src/main/scala/a/Tree.scala")
+  )
+
+  renamed(
+    "companion2",
+    """|/a/src/main/scala/a/Main.scala
+       |package a
+       |class <<Ma@@in>>{}
+       |object <<Main>>
+       |""".stripMargin,
+    newName = "Tree",
+    fileRenames =
+      Map("a/src/main/scala/a/Main.scala" -> "a/src/main/scala/a/Tree.scala")
+  )
+
+  renamed(
+    "many-files",
+    """|/a/src/main/scala/a/A.scala
+       |package a
+       |object A {
+       |  def <<ren@@ameIt>>(a : String) = ""
+       |}
+       |/a/src/main/scala/a/B.scala
+       |package a
+       |object B {
+       |  val str = A.<<renameIt>>("")
+       |}
+       |/a/src/main/scala/a/C.scala
+       |package a
+       |object C {
+       |  val str = A.<<renameIt>>("")
+       |}
+       |/a/src/main/scala/a/D.scala
+       |package a
+       |object D {
+       |  val str = A.<<renameIt>>("")
+       |}
+       |/a/src/main/scala/a/E.scala
+       |package a
+       |object E {
+       |  val str = A.<<renameIt>>("")
+       |}
+       |""".stripMargin,
+    newName = "iAmRenamed",
+    nonOpened = Set(
+      "a/src/main/scala/a/C.scala",
+      "a/src/main/scala/a/D.scala",
+      "a/src/main/scala/a/E.scala"
+    )
+  )
+
+  renamed(
+    "anon",
+    """|/a/src/main/scala/a/Main.scala
+       |trait Methodable[T] {
+       |  def <<method>>(asf: T): Int
+       |}
+       |
+       |trait Alphabet extends Methodable[String] {
+       |  def <<method>>(adf: String) = 123
+       |}
+       |
+       |object Main {
+       |  val a = new Alphabet {
+       |    override def <<me@@thod>>(adf: String): Int = 321
+       |  }
+       |}
+       |""".stripMargin,
+    newName = "renamed"
+  )
+
+  renamed(
+    "java-unchanged",
+    """|/a/src/main/java/a/Other.java
+       |package a;
+       |public class Other{
+       |
+       |}
+       |/a/src/main/scala/a/Main.scala
+       |package a
+       |object Main{
+       |  val other = new <<Oth@@er>>()
```

Comment:
Should we consider erroring in this case?

Reply:
We could, but the other way would be to use Metals to at least rename Scala occurrences and rename the rest by hand or via different LSP. 

I thinking maybe add a warning in that case?

Reply:
I added a warning in a case like that.